pkg/odo/util/completion/completionhandlers.go

```diff
@@ -39,16 +40,92 @@ var ServiceClassCompletionHandler = func(cmd *cobra.Command, args parsedArgs, co
 	completions = make([]string, 0)
 	services, err := context.Client.GetClusterServiceClasses()
 	if err != nil {
+		complete.Log("error retrieving services")
 		return completions
 	}
 
+	complete.Log(fmt.Sprintf("found %d services", len(services)))
 	for _, class := range services {
 		completions = append(completions, class.Spec.ExternalName)
 	}
 
 	return
 }
 
+// ServicePlanCompletionHandler provides completion for the the plan of a selected service
+var ServicePlanCompletionHandler = func(cmd *cobra.Command, args parsedArgs, context *genericclioptions.Context) (completions []string) {
+	completions = make([]string, 0)
+	if len(args.original.Completed) < 2 {
+		complete.Log("Couldn't extract the service name")
+		return completions
+	}
+
+	inputServiceName := args.original.Completed[1]
+
+	complete.Log(fmt.Sprintf("Using input: serviceName = %s", inputServiceName))
+
+	_, servicePlans, err := service.GetServiceClassAndPlans(context.Client, inputServiceName)
+	if err != nil {
+		complete.Log("Error retrieving details of service")
+		return completions
+	}
+
+	for _, servicePlan := range servicePlans {
+		completions = append(completions, servicePlan.Name)
+	}
+
+	return completions
+}
+
+// ServiceParameterCompletionHandler provides completion for the parameter names of a selected service and plan
+var ServiceParameterCompletionHandler = func(cmd *cobra.Command, args parsedArgs, context *genericclioptions.Context) (completions []string) {
+	completions = make([]string, 0)
+	if len(args.original.Completed) < 2 {
+		complete.Log("Couldn't extract the service name")
+		return completions
+	}
+
+	inputServiceName := args.original.Completed[1]
+	inputPlanName := args.flagValues["plan"]
+
+	complete.Log(fmt.Sprintf("Using input: serviceName = %s, servicePlan = %s", inputServiceName, inputPlanName))
+
+	_, servicePlans, err := service.GetServiceClassAndPlans(context.Client, inputServiceName)
+	if err != nil {
+		complete.Log("Error retrieving details of service")
+		return completions
+	}
```

Comment:
Can we try to share some code between the 2 completion handlers?

Reply:
Checking

Reply:
@metacosm I don't think this makes sense. 
Checking the updated handlers there is not much in common